### PR TITLE
net: lib: nrf_cloud: Include alerts source only when enabled

### PR DESCRIPTION
**Pull request by jtguggedal**
Include the nrf_cloud_alerts.c source file in the build only when alerts are enabled.

Signed-off-by: Jan Tore Guggedal <jantore.guggedal@nordicsemi.no>

Fixes failing CI

### CODE REVIEW

**Review by dilner — APPROVED** [review]

**Review by jorgenmk — APPROVED** [review]

**Comment by plskeggs** [comment]
I marked this DNM.  There was a reason I did it this way.  I do not know why the tests all passed before merge.

The goal was to allow samples and applications to call `nrf_cloud_alert_send()` or `nrf_cloud_rest_alert_send` without needing to wrap them in conditionals, per @jayteemo's opinion.  So with your change as it is, my changes to `nrf_cloud_mqtt_multi_service`  and `nrf_cloud_rest_device_message` (which were merged as part of my alerts PR already) will not compile if the customer disables alerts via Kconfig.

**Comment by jtguggedal** [comment]
> I marked this DNM. There was a reason I did it this way, but I can see it is problematic. I do not know why the tests all passed before merge.
> 
> The goal was to allow samples and applications to call `nrf_cloud_alert_send()` or `nrf_cloud_rest_alert_send` without needing to wrap them in conditionals, per @jayteemo's opinion. So with your change as it is, my changes to `nrf_cloud_mqtt_multi_service` and `nrf_cloud_rest_device_message` (which were merged as part of my alerts PR already) will not compile if the customer disables alerts via Kconfig.
> 
> Please update the two locations in `nrf_cloud_mqtt_multi_service/src/application.c` that call `nrf_cloud_alert_send()`, as well as the one location in `nrf_cloud_rest_device_message/src/main.c` that calls `nrf_cloud_rest_alert_send()` so the calls are also wrapped in `#if IS_ENABLED(CONFIG_NRF_CLOUD_ALERTS)` / `#endif`. Thanks.

@plskeggs If possible, it would be great if you (or someone else) could push a commit to my branch with those additions. Would be good to have this merged asap to unblock CI this close to the release

**Comment by glarsennordic** [comment]
An alternative approach is to #define the referenced functions in the header file as no-op macros if NRF_CLOUD_ALERTS is not defined. Then you don't need guards, and the .c file is still not compiled

I've seen this done in a few places, I'll see if I can dig up an example for precedent

File: subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c
Comment (glarsennordic) [inline subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c:261]: Could `nrf_cloud_alert_control_get` be protected in the same way as `nrf_cloud_rest_alert_send`?

Then you could either remove these guards, or if the intention is to keep them for clarity, replace them with `IS_ENABLED`

Comment (glarsennordic) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c:261]: Actually I'd probably protect nrf_cloud_alert_control_set too just for good measure, so that the entire module can be used in this manner

Comment (plskeggs) [reply subsys/net/lib/nrf_cloud/src/nrf_cloud_fsm.c:261]: I will consider that in a follow-on PR.  This needs to get merged to unblock CI.

**Review by plskeggs — APPROVED** [review]

File: include/net/nrf_cloud_alerts.h
Comment (plskeggs) [inline include/net/nrf_cloud_alerts.h:81]: Ah, I was unaware of this technique.  Good to know.

**Review by jayteemo — APPROVED** [review]

File: include/net/nrf_cloud_alerts.h
Comment (glarsennordic) [reply include/net/nrf_cloud_alerts.h:81]: Ooh, this is neat, and perfectly solves some issues we've been having!

**Pull request merged**
